recipes-gnome/librsvg/librsvg/0002-allow-invalid-ref-casting-to-allow-building.patch

```diff
@@ -0,0 +1,20 @@
+--- librsvg-2.52.10/src/surface_utils/mod.rs	2024-10-15 20:59:07.374334093 +0200
++++ librsvg-2.52.10/src/surface_utils/mod.rs_mod	2024-10-15 20:58:56.391383870 +0200
+@@ -147,12 +147,12 @@
+     #[inline]
+     fn set_pixel(&mut self, stride: usize, pixel: Pixel, x: u32, y: u32) {
+         let value = pixel.to_u32();
++        let index = (y as usize + stride) + (x as usize * 4);
+ 
+-        #[allow(clippy::cast_ptr_alignment)]
+-        unsafe {
+-            let p: *mut u8 = &mut self[y as usize * stride + x as usize * 4];
+-            *(p as *mut u32) = value;
+-        }
++        assert!(index + 4 <= self.len(), "Index out of bounds");
```

Comment:
This check is both incorrect and unnecessary.  copy_from_slice will do bounds checking.

Reply:
Thanks for this information. I didn't know that copy_from_slice do bounds checking, so I will remove this check.